src/modules/sdlog2/sdlog2.c

```diff
@@ -1822,6 +1847,31 @@ int file_copy(const char *file_old, const char *file_new)
 	return OK;
 }
 
+int check_free_space()
+{
+	/* use statfs to determine the number of blocks left */
+	FAR struct statfs statfs_buf;
+	if (statfs(mountpoint, &statfs_buf) != OK) {
+		warnx("could not determine statfs");
+		return ERROR;
+	}
+
+	/* use a threshold of 4 MiB */
+	if (statfs_buf.f_bavail < (int)(4*1024*1024/statfs_buf.f_bsize)) {
+		warnx("no more space on MicroSD (less than 4 MiB)");
+		mavlink_log_critical(mavlink_fd, "[sdlog2] no more space left on MicroSD");
```

Comment:
Here you don't do the `space_warning_sent` check like below. I know this works because `check_free_space()` is not called anymore. Maybe add a comment, as this is not immediately obvious for someone who just looks at this function.